core/domain/skill_services.py

```diff
@@ -1065,6 +1065,36 @@ def skill_has_associated_questions(skill_id):
     return len(question_ids) > 0
 
 
+def filtered_skills_with_none_type_keys(degrees_of_mastery):
+    """Given a dict of skills id's with there corresponding
+    mastery. It first split the dict and then returns a list of skill
+    id.
```

Comment:
Rather than explaining how this function works just explain what it does without mentioning the internal details.

Reply:
done!